internal/blob/download.go

```diff
@@ -0,0 +1,43 @@
+package blob
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/rs/zerolog/log"
+)
+
+// DownloadState downloads state data from blob storage.
+func DownloadState(ctx context.Context, urlstr string) ([]byte, error) {
+	log.Ctx(ctx).Debug().Msg("downloading state")
+
+	bucket, err := openBucket(ctx, urlstr)
+	if err != nil {
+		return nil, fmt.Errorf("error opening bucket: %w", err)
+	}
+	defer bucket.Close()
+
+	file, err := bucket.NewReader(ctx, "state.zst", nil)
+	if err != nil {
+		return nil, fmt.Errorf("error opening bucket file: %w", err)
+	}
+
+	zr, err := zstd.NewReader(file)
```

Comment:
from their docs: 
[It is important to use the "Close" function when you no longer need the Reader to stop running goroutines, when running with default settings.
](https://pkg.go.dev/github.com/klauspost/compress/zstd#section-readme:~:text=It%20is%20important%20to%20use%20the%20%22Close%22%20function%20when%20you%20no%20longer%20need%20the%20Reader%20to%20stop%20running%20goroutines%2C%20when%20running%20with%20default%20settings.)

Reply:
it's unconditionally closed now.